services/processor/native/adapter/native_compiler.go

```diff
@@ -202,6 +202,14 @@ func runGoCommand(ctx context.Context, workDir string, cmdArgs ...string) ([]byt
 		// "GOGC=off", (this improves compilation time by a small factor)
 	}
 	out, err := cmd.CombinedOutput()
+	if err != nil {
+		if deadline, ok := ctx.Deadline(); ok {
+			timeout := time.Until(deadline)
+			if timeout < 0 {
+				err = errors.Wrap(err, "after context deadline exceeded")
+			}
+		}
```

Comment:
> Why not just check `ctx.Err()`?

I am handling a specific case where context is timed out. `ctx.Err()` may be not nill in other cases.

The default message here was the text representation of `err`,  which was "signal: killed". I wanted to help the reader in the case of context timeout.

Reply:
I see you found a way to combine both error messages, nice. just to reiterate - you can reliably detect a timeout by checking for `ctx.Err() == context.DeadlineExceeded` it is however a little stinky, because it a little too specific. however, it is documented and apparent from the code that this condition will be true iff a timeout happened...

Reply:
thanks